### PR TITLE
Added metrics

### PR DESCRIPTION
**Pull request by stut**

### CODE REVIEW

File: pkg/v2/relay/cmd/server/main.go [hunk 1]
```diff
@@ -24,6 +26,9 @@ func main() {
 
 	var port uint = 0
 	flag.UintVar(&port, "port", 0, "the port to run on for the websocket")
+	var staticMetricsFilename string
```
Comment (ThePrimeagen) [inline pkg/v2/relay/cmd/server/main.go:29]: 2 things
1. move the code to a package pkg/v2/metrics
2. make it so that persistence is configurable (via some sort of interface)

the reason being is that i am going to want to create a persistence that sends down stats to my local machine every ~10 seconds or so

please respond to this right away if you work on it.  I am going to take care of it at some point and if you do it before me, awesome, if not, i'll do it :)

File: pkg/v2/relay/relay.go [hunk 2]
```diff
@@ -53,6 +63,19 @@ func (relay *Relay) Start() {
 		tmpl.ExecuteTemplate(w, "index.html", struct{}{})
 	})
 
+	http.HandleFunc("/metrics", func(w http.ResponseWriter, r *http.Request) {
```
Comment (ThePrimeagen) [inline pkg/v2/relay/relay.go:66]: i hate this file (as you see by the comments) but this is awesome

again with the metrics gathering i think that we could put this in metrics and have a "WriteTo(io.Writer) error" interface (something along those lines)

**Review by ThePrimeagen — CHANGES_REQUESTED** [review]

**Comment by ThePrimeagen** [comment]
as i said, i'll likely take care of it within 24 hours if you don't :)

so don't feel bad!!

its something i have been very interested in landing into the system

File: pkg/v2/relay/cmd/server/main.go (hunk 1 above)
Comment (stut) [reply pkg/v2/relay/cmd/server/main.go:29]: > please respond to this right away if you work on it. I am going to take care of it at some point and if you do it before me, awesome, if not, i'll do it :)

I've got a little time now so gonna have a crack at it.

File: examples/v2/doom/controller_test.go [hunk 3]
```diff
@@ -69,21 +69,22 @@ func TestDoomController(t *testing.T) {
 	dc.SendKey("e")
 	dc.SendKey("e")
 	require.Equal(t, []string{
-		"w", "a", "e",
+		"w", "a", "e", "e", "e", "e",
```
Comment (ThePrimeagen) [inline examples/v2/doom/controller_test.go:72]: this shouldn't happen

if there is a broken test, please let me know.  this was ... well its a long story :)

fixed in master.

if you could revert these changes and we can possibly add the rest.

File: pkg/v2/metrics/metrics.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+func New() *Metrics {
+	return &Metrics{
+		metrics:     make(map[string]int),
+		fileWriters: make([]*fileWriter, 0),
+	}
+}
+
+func (m *Metrics) WithFileWriter(filename string, format string, frequency time.Duration) *Metrics {
```
Comment (ThePrimeagen) [inline pkg/v2/metrics/metrics.go:35]: i would prefer to interface this out (i'll merge anyways)

the writer should just be an interface instead of hard coded.  i wanted to make a writer back to authorized sockets

Comment (stut) [reply pkg/v2/metrics/metrics.go:35]: No probs, can do. Sorry, misunderstood your comment about this.

Comment (stut) [reply pkg/v2/metrics/metrics.go:35]: No probs, will do. Sorry, I misunderstood your earlier message about this.
